### ISSUE TITLE
Android App Crashes while selecting the dropdown

### ISSUE THREAD

**Issue by AzadKshitij** [react-native-picker/picker#225]
App Crashes when trying to select dropdown from anywhere else than icon..

![61cf208bfa31d6c1c71d3f7b866ca05646b01c67](https://user-images.githubusercontent.com/54288405/111460099-79a64400-8713-11eb-900e-295a0ea4faf8.gif)

"react": "16.13.1"
"react-native": "0.63.3"
"@react-native-picker/picker": "^1.9.11"

**Comment by cenksari** [comment]
I confirm that: After updating React native v.0.64, all my applications crashes.

**Comment by sylvainbaronnet** [comment]
reported here already : https://github.com/react-native-picker/picker/issues/220

**Comment by jonra1993** [comment]
I am facing the same issue. Have you found any solution @AzadKshitij ?

"react": "17.0.1"
"react-native": "0.64.0"
"@react-native-picker/picker": "^1.12.0"

**Comment by AzadKshitij** [comment]
> I am facing the same issue. Have you found any solution @AzadKshitij ?
> 
> "react": "17.0.1"
> "react-native": "0.64.0"
> "@react-native-picker/picker": "^1.12.0"

Just lower down the version of react-native to 0.63.x

**Comment by jonra1993** [comment]
Thanks, @AzadKshitij in my case a carousel package was causing the problem you can find more information  on [#220](https://github.com/react-native-picker/picker/issues/220)

**Comment by VikBn** [comment]
"lower down the version of react-native to 0.63.x" is good solution, but what about version 0.64. Maybe somebody know, will it be fixed?

**Comment by rajeshlousigam** [comment]
in my case react-native-tab-view v3 was causing the problem, i download to react-native-tab-view 2.16.0, and it all work fine.

**Comment by BStenfors9119** [comment]
I am having this same issue:

Here are my dependencies in package.json:

```
"dependencies": {
    "@react-native-community/art": "^1.2.0",
    "@react-native-community/async-storage": "^1.12.0",
    "@react-native-community/masked-view": "^0.1.7",
    "@react-native-firebase/app": "^10.3.0",
    "@react-native-firebase/messaging": "^10.3.1",
    "@react-native-picker/picker": "^1.15.0",
    "@react-navigation/native": "^5.7.3",
    "@react-navigation/stack": "^5.2.9",
    "axios": "^0.19.2",
    "eslint": "^3.17.0",
    "moment": "^2.27.0",
    "moment-timezone": "^0.5.31",
    "prop-types": "^15.7.2",
    "react": "16.13.1",
    "react-icons": "^3.11.0",
    "react-native": "0.63.2",
    "react-native-device-info": "^2.3.2",
    "react-native-gesture-handler": "^1.7.0",
    "react-native-material-dialog": "^0.7.7",
    "react-native-material-ui": "^2.0.0-alpha.1",
    "react-native-progress": "^4.1.2",
    "react-native-restart": "0.0.20",
    "react-native-safe-area-context": "^3.1.7",
    "react-native-safe-area-view": "^1.1.1",
    "react-native-screens": "^2.4.0",
    "react-native-svg": "^12.0.3",
    "react-native-swipe-gestures": "^1.0.5",
    "react-native-tab-view": "^2.16.0",
    "react-native-vector-icons": "6.6.0",
    "react-navigation-material-bottom-tabs": "^2.2.8",
    "react-redux": "^5.1.2",
    "react-router-redux": "^4.0.8",
    "redux": "^3.7.1",
    "redux-form": "^7.4.3",
    "redux-form-material-ui": "^4.3.4",
    "redux-thunk": "^2.3.0",
    "underscore": "^1.10.2"
  }
```

I am noticing my version of React is a bit low.  I might try to upgrade that and see if it helps.

**Comment by Hassaan0** [comment]
Try linking **'@react-native-picker/picker'** manually

add below lines in **android>app>src>java>com>myApp>MainApplication.java**
`import com.reactnativecommunity.picker.RNCPickerPackage;`
add package in getPackages()
`packages.add(new RNCPickerPackage());`

in **android>app>build.gradle**
`implementation project('@react-native-picker/picker')`

**in android>settings.gradle** 
```
include '@react-native-picker/picker'
project('@react-native-picker/picker').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-picker/picker/android')
```

Hope it helps

**Comment by jafar-jabr** [comment]
?

**Comment by jafar-jabr** [comment]
> in my case react-native-tab-view v3 was causing the problem, I downgrade to react-native-tab-view 2.16.0, and it all works fine.

you save my day, thanks

**Comment by flyp-felype** [comment]
removing  `gestureEnabled:true`  from screenOptions  the solved my problem ->  https://github.com/react-native-picker/picker/issues/220#issuecomment-800196607

**Comment by iYahya** [comment]
i tried to open my picker using "ref" and it worked for me without any crash

```
 constructor(props) {
    super(props);
    this.pickerRef = createRef();
    }
    
    open() {
    this.pickerRef.current.focus();
  }

  close() {
    this.pickerRef.current.blur();
  }

render(){
return(
 <View style={{flex: 1, borderWidth: 1, borderColor: '#0f0'}}>
                    <TouchableWithoutFeedback onPress={() => this.open()}>
                      <View style={{width: wp(90), height: hp(5)}}>
                        <Text>LOL</Text>
                      </View>
                    </TouchableWithoutFeedback>
                    <View style={{opacity: 0}} pointerEvents={'none'}>
                      <Picker
                        mode={'dropdown'}
                        ref={this.pickerRef}
                        selectedValue={val}
                        onValueChange={(itemValue, itemIndex) =>
                          this.setState({val: itemValue})
                        }>
                        <Picker.Item label="1" value="1" />
                        <Picker.Item label="2" value="2" />
                        <Picker.Item label="3" value="3" />
                      </Picker>
                    </View>
                  </View>
)
}
```
so i customed a view to show my value and made my picker view  `pointerEvents={'none'}` to avoid getting that crash when clicking on picker

**Comment by rapomon** [comment]
This is my workaround without open manually the Picker:

```
            <View style={{flex: 1, borderWidth: 1, borderColor: '#0f0'}}>
              <Picker
                style={{...styles.itemPicker, color: theme.TEXT}}
                mode="dropdown"
                selectedValue={unit_elevation}
                onValueChange={(value) => this.changeUserPreferences('unit_elevation', value)}
              >
                <Picker.Item label={messages.meters} value="M" />
                <Picker.Item label={messages.feet} value="FT" />
              </Picker>
              <Text style={{width: '100%', height: 60, position: 'absolute', bottom: 0, left: 0}}>{' '}</Text>
            </View>
```

The trick consist in an empty <Text> component over the picker, it works fine.

**Comment by koladee** [comment]
Thanks @rapomon, you made my day!

**Comment by rapomon** [comment]
With the new release of react native 0.64.2 it's fixed.

**Comment by victorjordan95** [comment]
> This is my workaround without open manually the Picker:
> 
> ```
>             <View style={{flex: 1, borderWidth: 1, borderColor: '#0f0'}}>
>               <Picker
>                 style={{...styles.itemPicker, color: theme.TEXT}}
>                 mode="dropdown"
>                 selectedValue={unit_elevation}
>                 onValueChange={(value) => this.changeUserPreferences('unit_elevation', value)}
>               >
>                 <Picker.Item label={messages.meters} value="M" />
>                 <Picker.Item label={messages.feet} value="FT" />
>               </Picker>
>               <Text style={{width: '100%', height: 60, position: 'absolute', bottom: 0, left: 0}}>{' '}</Text>
>             </View>
> ```
> 
> The trick consist in an empty `<Text>` component over the picker, it works fine.

@rapomon solution worked for me! what a workaround! 😂 
Thanks a lot!!

**Comment by ejilee** [comment]
@rapomon Thank you! Also fixes a similar issue on a different library (react-native-picker-select)!

**Comment by Naturalclar** [comment]
This was a bug on React Native's side, should be fixed on React Native 0.64.2
